actions.js

```diff
@@ -55,3 +57,128 @@ export function constructContactItemsWithToken(token) {
     });
   };
 }
+
+export function getHeyMentorToken(token, authType) {
+  return async (dispatch) => {
+    dispatch({
+      type: 'SET_USER',
+      data: {
+        loading: true,
+      }
+    });
+    const response = await fetch(
+      `${API_URL}/register/${authType}?access_token=${token}`,
+      { method: 'post' }
+    );
+
+    try {
+      const responseJson = await response.json();
+      if (responseJson && !responseJson.error) {
+        dispatch({
+          type: 'SET_USER',
+          data: {
+            hmToken: responseJson,
```

Comment:
> We should parse this response and store the exact things we need for the app, instead of saving the whole blob returned from the API (similar to what we do with the response for the contact data). The app logic should be based on storage, as opposed to based on the API, that way we can alter the API more cleanly.

I have made change to only take the `_id` and `user_type` attributes from `responseJson`

 
Regarding the second part, I didn't get what you meant. Can you please elaborate by giving an example?

> It also looks like the redux persist logic only compares to an object depth of 1, which is a limitation on the storage that we don't have on the API, so the storage function should handle that constraint, and not the API.

Reply:
The second part was just additional motivation for us parsing the API blob and saving it to our own structure in redux. Persist only looks 1 level deep in your object when deciding how to merge ([doc](https://github.com/rt2zz/redux-persist#basic-usage)).